wireless_modules/sensors/gps_sensor.py

```diff
@@ -0,0 +1,61 @@
+from machine import UART
+import uasyncio as asyncio
+
+from micropyGPS import MicropyGPS
+
+from sensor_base import Sensor
+
+
+class GpsSensor(Sensor):
+    """
+    A GPS sensor implementing NMEA-0183 connected via UART, such as the u-blox NEO 6M.
+    """
+
+    def __init__(self, uart_channel: int):
+        """
+        Begin reading data from the GPS over UART.
+        :param uart_channel: The UART channel that the GPS is connected to.
+        """
+        self.gps = MicropyGPS(location_formatting="dd")
+        asyncio.create_task(self.uart_rx(uart_channel))
+
+    async def uart_rx(self, uart_channel: int):
+        """
+        Read and load data from the GPS over UART.
+        """
+        uart = UART(uart_channel, baudrate=9600)
+        stream_reader = asyncio.StreamReader(uart)
+        while True:
+            res = await stream_reader.readline()
+            for char in res:
+                self.gps.update(chr(char))
+
+    def read(self):
+        """
+        Get the most recently stored GPS data.
+        :return: An array containing the GPS sensor dictionary. `value` in this
+        dictionary contains spacial, quality of service and time information.
+        """
+        latitude = self.gps.latitude[0] * (1 if self.gps.latitude[1] == "N" else -1)
+        longitude = self.gps.longitude[0] * (1 if self.gps.longitude[1] == "E" else -1)
```

Comment:
I'm assuming that the N and E correspond to north and east. Is there any chance you could add an inline comment to clear this up?